src/vec_bcd_ppc.h

```diff
@@ -461,6 +556,244 @@ vec_cbcdaddcsq (vBCD_t *cout, vBCD_t a, vBCD_t b)
+ * partial products.
+ * Following code completes the implementation of BCD multiply low:
+ * \code
+  ...
+  else
+    {
+      _Decimal128 d_ah, d_bh, d_hl, d_lh, d_h;
+
+      high_a = vec_bcdsrqi (a, 16);
+      high_b = vec_bcdsrqi (b, 16);
+
+      d_ah = vec_BCD2DFP (high_a);
+      d_bh = vec_BCD2DFP (high_b);
+
+      d_hl = d_ah * d_b;
+      d_lh = d_a * d_bh;
+
+      d_h = d_hl + d_lh;
+      d_h = __builtin_dscliq (d_h, 17);
+      d_h = __builtin_dscriq (d_h, 1);
+
+      d_t = d_p + d_h;
+    }
+ * \endcode
+ * Here we know that there are higher order digits in one or both operands.
+ * First use vec_bcdsrqi() to isolate the high 15-digits of operands a
+ * and b. Both Vector unit and DFU have decimal shift operations,
+ * but the vector shift operation is faster.
+ *
+ * Then convert to DFP and
+ * multiply (high_a * low_b and high_b * low_a) for the two middle
+ * order partial products which are summed. This sum represents the
+ * high 32-digits (the 31-digit sum can carry) of a 48-digit product.
+ * Only the lower 16-digits of this sum is needed for the final sum and
+ * this needs to be aligned with the high 16 digits of the original
+ * lower order partial product.
+ *
+ * For this case use
+ * <B>DFP Shift Significand Left Immediate</B> and
+ * <B>DFP Shift Significand Right Immediate</B>. All the data is in
+ * the DFU and the high cost of the DFU shift is offset by avoiding
+ * extra format conversions. We use shift left 17 followed by shift
+ * right 1 to clear the highest order DFP digit and avoid any overflow.
+ * A final DFP add produces the low order 32 digits of the product
+ * which will be truncated to 31-digits in the conversion to BCD.
+ *
+ * How we can look at the BCD multiply high (generate the full 62-digit
+ * product returning the high 31 digits) and point out the differences.
+ * Multiply high also start by isolating the low order 16 BCD
+ * digits, performing he low order multiply (low_a * low_b),
+ * and testing for the short circuit (all higher order digits are 0).
+ * The first difference (from multiply low) is that in this case
+ * only the high digit of the potential 32-digit product is returned.
+ *
+ * \code
+  const vBCD_t dword_mask = (vBCD_t) CONST_VINT128_DW(15, -1);
+  vBCD_t t, low_a, low_b, high_a, high_b;
+  _Decimal128 d_p, d_t, d_a, d_b;
+
+  low_a = vec_and (a, dword_mask);
+  low_b = vec_and (b, dword_mask);
+  d_a = vec_BCD2DFP (low_a);
+  d_b = vec_BCD2DFP (low_b);
+  d_p = d_a * d_b;
+  if (__builtin_expect ((vec_cmpuq_all_eq ((vui128_t) low_a, (vui128_t) a)
+      && vec_cmpuq_all_eq ((vui128_t) low_b, (vui128_t) b)), 1))
+    {
+      d_t = __builtin_dscriq (d_p, 31);
+    }
+  else
+    {
+    ...
+    }
+  t = vec_DFP2BCD (d_t);
+ * \endcode
+ * So the short circuit code shifts the low partial product right 31
+ * digits and returns that value.
+ *
+ * If we can not short circuit Multiply high requires  the generation
```

Comment:
need some separation after "short circuit", if I understand the sentence correctly.  Maybe simply a comma after "short circuit".

Reply:
OK